### ISSUE TITLE
Fancier calibration

### ISSUE THREAD

**Issue by davidbenjamin** [davidbenjamin/permutect#38]
Calibration is currently temperature scaling.  That is, if the uncalibrated model emits a logit output y, the calibrated model emits y/T for some temperature T that is fit to the validation set by max likelihood.  In the current version T depends on the alt read count only.

The problem with this is that it can't account for extreme overconfidence.  A better approach would some mapping from uncalibrated logits to calibrated logits that saturates.  Realistically, regardless of the uncalibrated confidence there is a possibility of completely missing some type of artifact (or just of plain old overfitting, of course) and so there needs to be a ceiling on the calibrated confidence.

One way to do this would be a tanh eg calibrated logit = A * tanh(B * uncalibrated logit).  Note that although this is a tanh the output is still a logit, not an output probability.  It only needs a few more parameters than temperature scaling and so is still safe to fit to the validation set, although obviously we must not compromise the test set.
